File: observer/c-sharp/ObserverUser.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Observer
{
    // Observer arayüzünü uygulayan, değişimi takip eden sınıflardır.
    // Değişim update metotu ile bildirilir.
    // UML diyagramındaki ConcreteObserver yapısına denk gelmektedir.
    class ObserverUser : IObserverUser
    {
        public void SendNotification(ProductUpdateMessage message)
        {
            Console.WriteLine(message.ToString());
        }
    }
}

File: observer/go/observer-user.go
package main

import "fmt"

// Observer arayüzünü uygulayan, değişimi takip eden sınıflardır.
// Değişim update metotu ile bildirilir.
// UML diyagramındaki ConcreteObserver yapısına denk gelmektedir.
type observerUser struct{}

func NewObserverUser() *observerUser {
	return &observerUser{}
}

func (u observerUser) SendNotification(msg ProductUpdateMessage) {
	fmt.Println(msg.ToString())
}
